sparse/src/KokkosSparse_SellMatrix.hpp

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright Contributors to the Kokkos project
+
+/// \file KokkosSparse_SellMatrix.hpp
+/// \brief Local sparse sliced ellpack matrix interface
+///
+/// This file provides KokkosSparse::SellMatrix.  This implements
+/// a parse matrix stored in sliced ellpack ("Sell") format.
+
+#ifndef KOKKOSSPARSE_SELLMATRIX_HPP_
+#define KOKKOSSPARSE_SELLMATRIX_HPP_
+
+#include "Kokkos_Core.hpp"
+
+namespace KokkosSparse {
+namespace Experimental {
+
+/// \class SellMatrix
+/// \brief Sliced ell format implementation of a sparse matrix.
+/// \tparam ScalarType The type of entries in the sparse matrix.
+/// \tparam OrdinalType The type of column indices in the sparse matrix.
+/// \tparam Device The Kokkos Device type.
+/// \tparam MemoryTraits Traits describing how Kokkos manages and
+///   accesses data.  The default parameter suffices for most users.
+///
+/// "Crs" stands for "compressed row sparse."  This is the phrase
+/// Trilinos traditionally uses to describe compressed sparse row
+/// storage for sparse matrices, as described, for example, in Saad
+/// (2nd ed.).
+template <class ScalarType, class OrdinalType, class Device, class MemoryTraits = void,
+          class SizeType = KokkosKernels::default_size_type>
+class SellMatrix {
+  static_assert(std::is_signed<OrdinalType>::value, "CrsMatrix requires that OrdinalType is a signed integer type.");
+
+ private:
+  using host_mirror_space = typename Kokkos::ViewTraits<ScalarType*, Device, void, MemoryTraits>::host_mirror_space;
+
+ public:
+  //! @name Typedefs
+  //@{
+
+  //! Type of the matrix's execution space.
+  using execution_space = typename Device::execution_space;
+  //! Type of the matrix's memory space.
+  using memory_space = typename Device::memory_space;
+  //! Canonical device type
+  using device_type = Kokkos::Device<execution_space, memory_space>;
+
+  //! Type of each value in the matrix.
+  using value_type = ScalarType;
+  //! Type of each (column) index in the matrix.
+  using ordinal_type  = OrdinalType;
+  using memory_traits = MemoryTraits;
+  /// \brief Type of each entry of the "row map."
+  ///
+  /// The "row map" corresponds to the \c ptr array of row offsets in
+  /// compressed sparse row (CSR) storage.
```

Comment:
```suggestion
```
I think this is not relevant here (copy paste from CrsMatrix.hpp) but maybe it still is?